kombu/utils/url.py

```diff
@@ -38,6 +44,24 @@ def parse_url(url):
     # type: (str) -> Dict
     """Parse URL into mapping of components."""
     scheme, host, port, user, password, path, query = _parse_url(url)
+    if query:
+        keys = list(query.keys())
```

Comment:
Why do we need to make a list out the keys?

Reply:
In python 3 `.keys()` is a generator so if I don't make it a list then I'm iterating over the dictionary and modifying it at the same time (line 63) which would cause a warning and is dangerous activity.